tests/upgrades/test_yum_plugins.py

```diff
@@ -0,0 +1,258 @@
+        product.sync()
+        repolist = [tools_repo, rhel_repo]
+        content_view = self._publish_content_view(org=org, repolist=repolist)
+
+        ak = entities.ActivationKey(content_view=content_view, organization=org.id,
+                                    environment=environment).create()
+        subscription = entities.Subscription(organization=org).search(query={
+            'search': 'name={}'.format(product.name)})[0]
+
+        ak.add_subscriptions(data={'subscription_id': subscription.id})
+        rhel7_client = dockerize(ak_name=ak.name, distro='rhel7', org_label=org.label)
+        client_container_id = [value for value in rhel7_client.values()][0]
+        client_container_name = [key for key in rhel7_client.keys()][0]
+        self._host_location_update(client_container_name=client_container_name, loc=loc)
+
+        status = execute(docker_execute_command, client_container_id,
+                         'subscription-manager identity', host=self.docker_vm)[self.docker_vm]
+        self.assertIn(org.name, status)
+
+        self._install_or_update_package(client_container_id, 'katello-agent')
+        self._run_goferd(client_container_id)
+
+        scenario_dict = {self.__class__.__name__: {
+            'rhel_client': rhel7_client,
+            'product_id': product.id,
+            'conten_view_id': content_view.id
+        }}
+        create_dict(scenario_dict)
+
+    @post_upgrade
+    def test_post_scenario_yum_plugins_count(self):
+        """Upgrade katello agent on pre-upgrade content host registered
+        with Satellite.
+
+        :id: postupgrade-45241ada-c2c4-409e-a6e2-92c2cf0ac16c
+
+        :steps:
+            1. Create custom tool repo as per upgraded Satellite version and sync.
+            2. Add in content-view and publish it.
+            2. Update Katello-agent and Restart goferd.
+            2. Check yum plugins count.
+
+        :expectedresults:
+            1. Loaded yum plugins should not load more than two times.
+         """
+        entity_data = get_entity_data(self.__class__.__name__)
+        client = entity_data.get('rhel_client')
+        client_container_id = list(client.values())[0]
+        product_id = entity_data.get('product_id')
+        conten_view_id = entity_data.get('conten_view_id')
+        product = entities.Product(id=product_id).read()
+        content_view = entities.ContentView(id=conten_view_id).read()
+
+        tools_repo, rhel_repo = self._create_custom_rhel_tools_repos(product)
+        product.sync()
+        for repo in (tools_repo, rhel_repo):
+            content_view.repository.append(repo)
+        content_view = content_view.update(['repository'])
+        content_view.publish()
+
+        self._install_or_update_package(client_container_id, "katello-agent", update=True)
+        self._run_goferd(client_container_id)
+        self._check_yum_plugins_count(client_container_id)
```

Comment:
What is the difference between _check_yum_plugins_count in pre and post-upgrade step? What do you think about adding the same step in pre-upgrade as well?

Reply:
@omkarkhatavkar yeah it only requires to check at upgraded(post upgrade) version to see such[1] type of issue if any in upgraded package/tool: 
[1] https://bugzilla.redhat.com/show_bug.cgi?id=1625649

Reply:
My intention was raising this, what if loading plugin issue itself exist in pre-upgrade. There is no way that you could tell that it was working in pre-upgrade.

Reply:
@omkarkhatavkar this time bug is fixed for 6.5.